ADR/00xx-component-references.md

```diff
@@ -0,0 +1,147 @@
+
+---
+
+### OLM Operators, with components in the same repo
+
+Another OLM Operator Scenario: an operator git repo contains both the controller code and the bundle metadata.
+
+* The user has two Components, that both point to the same repo.
+* The bundle Component declares that it embeds references to the controller image Component, by way of a new `ComponentReference` CR.
+* [integration-service] will always skip testing for the controller component because it is a *referent* Component.
+* [build-service] will propagate the digest reference as a PR to the bundle image, which happens to be the same git repository as the controller Component.
+  * The user submitted their controller image update on the `new-feature` branch.
+  * [build-service] could do one of the following:
+    * ❌ Push its commits with the new image digest references to the same feature branch.
+      * Will [build-service] -- acting as the TAP github App -- have rights to push to that branch? Probably not.
+    * 🤔 Submit its commits as a suggestion on the original PR, which the user can accept or not.
+      * This would be “odd” for build service, but maybe nice from the user’s point of view. Just one PR to work with.
+    * ❌ Push its commits with the new image digest references to a new feature branch based on the original feature branch, called `new-feature-build-service`
+      * This could be nice, but the PR will trigger new builds of the original image since it contains code changes for the controller (here, I am assuming that we can limit builds in PaC based on contextDirectories for multi-Component repos; we don’t have this today, but assume that we do).
+    * ✅ Push its commits with the new image digest references to a new feature branch that is not based on the original feature branch, called `new-feature-from-build-service`
+      * This works like it does in most other cases. Let the user merge the original PR, and only after that will [integration-service] update the checks for `new-feature-from-build-service` to say “okay to merge now!”
+
+## Open Questions
+
+* To give the user some control, should we make [build-service] respect [renovatebot
+  configuration files](https://docs.renovatebot.com/configuration-options/)? Does that still make
+  sense if renovatebot is not being run on a cron basis? (If the answer to the original question is
+  yes, then that may inform whether how we implement the new functionality in [build-service].
+* Since we don't have exclusive control over merging changes (or really, any control at all
+  exclusive or not), we can't prevent the user from merging things out of order. This means they can
+  get themselves into trouble. We try to provide feedback via github checks to help prevent this;
+  but we can't stop it. How does the user recover from this situation? Are there any side-effects?
```

Comment:
I don't foresee that this is really an issue. AppStudio will have to fall back on Snapshot consistency checks to ensure that all _necessary_ references within the snapshot are up to date. If the images are merged out of order and the image digests from the latest PR are reused, then the snapshot will be in sync quickly. Worst case (i.e. if merge commits are created), Renovatebot will eventually come around and update the references but this _should_ "just work" as the two produced containers should be functionally identical.